### ISSUE TITLE
Select by PK definition on a view, not generating corresponding root query

### ISSUE THREAD

**Issue by mvid** [hasura/graphql-engine#6004]
I have a view that I am exposing for authorization reasons, and while Hasura is generating the root query for `Select`, it isn't for `Select by PK`, even though it is accepting the definition in the config. 

Does Hasura not support primary key single object lookups on view? If not, why is it a config option? If so, why isn't it being generated?

Thanks,
Mantas

**Comment by tirumaraiselvan** [comment]
What do you mean by `even though it is accepting the definition in the config` ?

Views in Postgres cannot have constraints like primary key, unique etc. Hence `select_by_pk` is not generated for views.

**Comment by mvid** [comment]
When you go to the Modify tab on a view, it allows you to set the name for a Select One query. Why is that UI element there if it doesn't generate anything?

**Comment by tirumaraiselvan** [comment]
Indeed, it might be confusing to see the "name customization" for `Select One` query for Views. It should ideally be hidden but that config option is nothing more than a "new name" for an auto-generated field.

**Comment by AndrewCraswell** [comment]
Hmm I'm hitting this scenario with MS SQL. I have a table with fields, and a View which augments some of the data with static calculations. Like, creating a `full_name` column by joining `first_name` and `last_name` so that text searching will work better.

Unfortunately, this seems to mean that the "Select by PK" query is never generated, so even though the underlying table has a primary key, Hasura assumes there cannot be a select by Id query generated.

Ideally I wouldn't need to use a view for my scenario at all, but I wasn't able to find a workaround...

**Comment by gigablack** [comment]
Maybe you can use SQL functions for that https://hasura.io/docs/latest/schema/postgres/custom-functions/
